### ISSUE TITLE
Caps_lock modification inches from working:)

### ISSUE THREAD

**Issue by Jeroen2000** [pqrs-org/Karabiner-Elements#2085]
Hi all,

Beginner needing a bit of help on this one. 

tl;dr:
Default MacOS situation => with caps_lock == ON => when I hit the key with number '1' , I get '&'. 
Desired situation => with caps_lock == ON => when I hit the key with number '1' , I get '1'. 
Current situtation => with caps_lock == ON => when I hit the key with number '1' , I get '1' _but_ **caps_lock gets disable after pressing '1'** Why is caps_lock getting disabled. Why is the key down event in bold being sent?

> type:key_down        code:225        name:left_shift      misc : flags: left_shift 
> type:key_down        code:30         name:1               misc : flags: left_shift 
> type:key_up          code:225        name:left_shift      misc:
> **type:key_down        code:57         name:caps_lock       misc:**
> type:key_up          code:30         name:1               misc:
> 
 code for digit '1':

{ 
          "type":"basic",
          "from": 
           {
            "key_code":"1",
            "modifiers": 
            {
              "mandatory":["caps_lock"],
              "optional":[]
            }
          },
          "to":[
            { 
              "key_code":"1",
               "hold_down_milliseconds": 100,
               "modifiers":["left_shift"]
           }    
          ]

**Comment by Jeroen2000** [comment]
@tekezo 
Here is my working code for the issue above. Note I just added "caps_lock" to "modifiers" in the "to" section. Every time I type a number, my led blinks. So I believe I may have stumbled onto a "bug" or `inconsistency. I'd be grateful if you can take a look whenever you find the time.
`
{
  "title": "Jeroen AZERTY better caps_lock",
  "rules": [
    {
      "description": "Enabled caps_lock (geen led lit) gives you numbers on AZERTY keyboard. You lose upper case é, à and ç",
      "manipulators": [
        {
          "type":"basic",
          "from": {
            "key_code":"1",
            "modifiers": {
              "mandatory":["caps_lock"],
              "optional":[]
            }
          },
          "to": [
        {
           "key_code":"1",
            "modifiers":["caps_lock","left_shift"]
        }
    ]
        },
        {
          "type":"basic",
          "from": {
            "key_code":"2",
            "modifiers": {
              "mandatory":["caps_lock"],
              "optional":[]
            }
          },
          "to":[
            {
              "key_code":"2",
               "modifiers":["caps_lock","left_shift"]
           }
         
    ]
        },
        {
          "type":"basic",
          "from": {
            "key_code":"3",
            "modifiers": {
              "mandatory":["caps_lock"],
              "optional":[]
            }
          },
          "to":[
            {
              "key_code":"3",
               "modifiers":["caps_lock", "left_shift"]
           }
         ]
        },
        {
          "type":"basic",
          "from": {
            "key_code":"4",
            "modifiers": {
              "mandatory":["caps_lock"],
              "optional":[]
            }
          },
          "to":[
            {
              "key_code":"4",
               "modifiers":["caps_lock", "left_shift"]
           }
         ]
        },
        {
          "type":"basic",
          "from": {
            "key_code":"5",
            "modifiers": {
              "mandatory":["caps_lock"],
              "optional":[]
            }
          },
          "to":[
            {
              "key_code":"5",
               "modifiers":["caps_lock", "left_shift"]
           }
         ]
        },
        {
          "type":"basic",
          "from": {
            "key_code":"6",
            "modifiers": {
              "mandatory":["caps_lock"],
              "optional":[]
            }
          },
          "to":[
            {
              "key_code":"6",
               "modifiers":["caps_lock", "left_shift"]
           }
         ]
        },
        {
          "type":"basic",
          "from": {
            "key_code":"7",
            "modifiers": {
              "mandatory":["caps_lock"],
              "optional":[]
            }
          },
          "to":[
            {
              "key_code":"7",
               "modifiers":["caps_lock", "left_shift"]
           }
         ]
        },
        {
          "type":"basic",
          "from": {
            "key_code":"8",
            "modifiers": {
              "mandatory":["caps_lock"],
              "optional":[]
            }
          },
          "to":[
            {
              "key_code":"8",
               "modifiers":["caps_lock", "left_shift"]
           }
         ]
        },
        {
          "type":"basic",
          "from": {
            "key_code":"9",
            "modifiers": {
              "mandatory":["caps_lock"],
              "optional":[]
            }
          },
          "to":[
            {
              "key_code":"9",
               "modifiers":["caps_lock", "left_shift"]
           }
         ]
        },
        {
          "type":"basic",
          "from": {
            "key_code":"0",
            "modifiers": {
              "mandatory":["caps_lock"],
              "optional":[]
            }
          },
          "to":[
            {
              "key_code":"0",
               "modifiers":["caps_lock", "left_shift"]
           }
         ]
        }


      ]
    }
  ]
}

**Comment by kn666** [comment]
I've come to the same problem and this looks like a bug. If you check key codes generated, there are many weird caps_lock downs and ups.

**Comment by kn666** [comment]
I use this combination for now (from goku). 'a' may be replaced with 'grave_accent_and_tilde' to stay untouched:
```
        "description" : "base",
        "manipulators" : [ {
          "from" : {
            "key_code" : "grave_accent_and_tilde"
          },
          "to" : [ {
            "key_code" : "a"
          } ],
          "type" : "basic"
        }, {
          "from" : {
            "key_code" : "grave_accent_and_tilde",
            "modifiers" : {
              "optional" : [ "caps_lock" ]
            }
          },
          "to" : [ {
            "key_code" : "b"
          } ],
          "type" : "basic"
        },```

**Comment by jrop** [comment]
I've hit the same issues described in this thread as well.  I am wanting to map CapsLock to "Esc" when pressed alone, but also have it as a modifier for my `hjkl` keys and map them to left, down, up, and right arrow keys, respectively.  I got close, but couldn't get it completely working.

**Comment by jrop** [comment]
I ended up using this: https://github.com/Vonng/Capslock
It is working for me, but I have not dug deeper to see how they solved this issue (under the hood it is using Karabiner Elements)
